### ISSUE TITLE
Offer alternative reactions to "like" a post, similar to facebook reactions

### ISSUE THREAD

**Issue by mangostaniko** [diaspora/diaspora#6766]
Facebook now offers alternative ways to react to posts: 
["like", "love",  "haha", "yay", "wow", "sad" and "anger"](http://mashable.com/2015/10/08/facebook-reactions-how-to/#cUrq6KAeOqqW)

Why is this important? Textual global communication today is highly prone to be ambiguous and lead to misunderstandings and thus being unable to apply empathy. More faces / smileys / concise description of how one feels about a post can help a lot to create a better atmosphere in online communication.

**Comment by abimelex** [comment]
I guess there is nothing more important than this! Who cares about post after editing or comment likes? What we need is to clone of the newest Facebook features!

**Comment by denschub** [comment]
@abimelex There is no need to attack a feature request.

**Comment by Zauberstuhl** [comment]
has probably the same issue like the comment-like-thingy .. federation madness for small pods

**Comment by Clarin98** [comment]
Thanks for your feature request @mangostaniko. For my part, I don't believe that the feature be strictly necessary, could be useful for trolls and bad management of content into the community :)

**Comment by mangostaniko** [comment]
Yes i also think it should not be viewed as a priority since there is a lot of other things to do, but i deem it useful as it allows for more accurate expression of emotional reactions. It has always struck me how much misunderstandings occur all around on social networks and how sometimes there is rather limited ability for empathy due to the highly text based communication and i think this would help a bit at least. gifs and memes are probably better though ;)

**Comment by stragu** [comment]
@Zauberstuhl Kind of off-topic, but is the "federation madness for small pods" -issue likely to be something that will change (fixed/enhanced) in the future, or is it something that the federation will have to live with because that's how it is designed to start with?

**Comment by Zauberstuhl** [comment]
@stragu every "like" and in this case every "yay", "wow" .. will be federated and requires some space in the database. At the first glance that looks like nothing but if you run a server for couple of months (even if you have a small user-base) that will fill-up the db, cause your small pod with almost no local-user activity will receive every public "yay" from all servers world-wide.

currently this is a design issue; the fix would be not federate such events.. which would negate this feature for smaller pods. Currently some dudes <3 are polishing and improving the federation protocol but this problem is different. I do not expect that it will be fixed in the near future..

**Comment by kephas** [comment]
But don't small pods already receive all likes? What's the difference in load to receive yays and angers instead?

**Comment by SuperTux88** [comment]
The difference is, that one can only like once per post, but could react multiple times with different "yays and angers". But back in 2016 the `likes` table was a lot bigger than now, because it contained all signatures for all likes (which was also the reason why likes for comments were removed a long time ago). Now (after the federation refactoring) only the necessary signatures are stored, which makes the `likes` table a lot smaller.

**Comment by goobertron** [comment]
Pleeeease can we not be like Facebook? Facebook is killing conversation and normal human interactions. I would really like Diaspora to remove 'likes' completely and let us talk to each other rather than making facile gestures.

**Comment by SuperTux88** [comment]
It's not about being facebook or not (Ok, the issue-text mentions facebook, but I think we could have the same issue without mentioning facebook). Sometimes a simple like or reaction can be enough or even better than comments. For example we had this once with a user who started sending "thumbs-up-image"-comments, because likes weren't federated correctly from hubzilla to diaspora. This was very annoying, because for a comment everybody gets a notification, while for a like only the author gets a notification (and you could theoretically do the same with other reaction-images now). And even a small number of reactions like here on github are already enough many times to express what you think, without annoying every watcher with notifications (while a "like" sometimes doesn't match).

So while I see your point, that conversations are important, I sometimes just don't react at all, because I don't want to comment, and "like" doesn't match. So don't know if it would be an advantage or a disadvantage to have more options. But maybe adding likes for comments back is just enough, so I can like reactions of others which match my opinion?

So I'm really undecided here (and that's probably a reason why there is a "needs discourse" label here), and this definitely has no priority at all for me, I'm not completely against it. However, all I wanted to say with my last comment was, that it's maybe more possible now, than it was back in 2016, but it's still "needs discourse" (and if we add it, we should add it, because we think it could be useful, and not because of facebook).

**Comment by goobertron** [comment]
Hi @SuperTux88, and thanks for your detailed reply.

My comment was not in response to yours, but to those further up the discussion, including the OP, which could certainly be interpreted as 'Facebook does x, therefore Diaspora should do x'.

I appreciate your points about the benefits of non-comment reactions. (I sometimes use the 'reaction' button here in Github because I know that all repo owners get a notification for every comment, and I want to avoid bothering them.) And I realise that human interaction face to face involves a lot more than mere words, and that it would be good to replicate this in some way in on-screen interactions. But the 'like' has unfortunately devalued interaction to such an extent that it seems to have become in many cases a meaningless clicking of buttons as an alternative to actually interacting with other people.

If likes and other click-button responses are too valuable to the majority of people to consider removing them, I'll repeat my suggestion to make it possible for a user to to able to disable likes on their posts.

As I'm now straying further from what's useful on this Github ticket, let's take this to Discourse as you suggested.

**Comment by kephas** [comment]
@SuperTux88, on no social network that I know can you have several reactions to the same object. So no, it wouldn't change the load. If you react with "yay", you cannot also react with "anger".

As for the effect on discussion, it is the same as in many physical meetings: humans express their reactions to a discussion without participating in it verbally, as a background noise, and it is a valuable tool for us social animals. We nod, we make small (or grand) gestures, etc…

It hasn't killed discussion AFK, not sure why it would in social networks.

**Comment by goobertron** [comment]
> humans express their reactions to a discussion without participating in it verbally, as a background noise, and it is a valuable tool for us social animals. We nod, we make small (or grand) gestures, etc…

I know that this is what these emoji are intended to mimic, but their effect on human interaction and the individual psyche is not at all similar to physical gestures. Do you really feel exactly the same when you see 'Alice has reacted to your post: 🤗' in your notifications as when someone physically gives you a hug? No, of course not.

I'd say, contrary to what you say, that body language and gestures are not 'background noise' of conversations but actually carry the weight of interaction during verbal conversations, and convey a lot of meaning. We've been programmed to respond to those gestures over millions of years. Little yellow cartoons (or whatever particular form they take) on a computer screen can't possibly affect us in the same way.

As to their effect on conversations, I find in real life that gestures form an integral part of conversations rather than replacing them. On social media, I find that probably more than 75% of the interaction I get from other people on my posts is 'likes' or other available 'one-click reactions', including resharing public posts on Diaspora, without any verbal interaction at all. It's clear that people have become conditioned, or have consciously chosen, to use one-click options instead of actually interacting in any more involved way. In many cases this is likely to be because they're trying to interact with as many posts as possible from as many people as possible, which is something that doesn't happen in physical interactions. As far as I am concerned, if not actually killing verbal interaction (because some does happen, sometimes at least) it has had the effect of severely hampering it.

I'd actually be happy for Diaspora to enable as many of these one-click 'reactions' as they want, as long as each user is given the option of enabling or disabling them for their own posts/content. And I'd ideally like this to be extended to the existing 'like' function. Is there any reason that this would not be possible/viable?

**Comment by kephas** [comment]
You make a lot of assumptions about people's motives, and seem to generalize a lot from a possibly very informal and probably biased evaluation of your personal experience.

Maybe that's not a great basis to reject a UI element…

**Comment by stragu** [comment]
@kephas , I don't think @goobertron is altogether rejecting the idea, he's proposing to make it an optional element. I understand why some users might want to do away with "one-click reactions" ("OCRs"?) by unticking a box in their settings, that option would make a nice middle-ground. If someone thinks this should absolutely not be implemented, I'd like them to back it up with some published scientific research showing the negative effects of those OCRs on the quality of conversation.

As far as "becoming like Facebook" goes, I don't think we should reject anything on the basis that Facebook (or whatever other social network) has already implemented it. We already share quite a few elements with their software, and there are probably lots of great other ideas in there that we should also add. However, it is a very healthy thing to first ponder what could the effects of a new function be, like people are doing here.

**Comment by goobertron** [comment]
> I don't think we should reject anything on the basis that Facebook (or whatever other social network) has already implemented it.

No, we definitely shouldn't reject any feature on the basis that some other service has implemented it; but we also shouldn't consider it important to implement any particular feature simply because one of the biggest networks has done so (and therefore 'people expect it'), which is what I said above. We should always think, when considering each individual feature, about what is best for Diaspora.

> You make a lot of assumptions about people's motives, and seem to generalize a lot from a possibly very informal and probably biased evaluation of your personal experience.

I'm giving a considered analysis based of my experience of using social networks over 10+ years, plus reports and analyses by other people that I've read. What are you doing, if not that?
